core/drivers/gic.c

```diff
@@ -858,6 +858,30 @@ void gic_dump_state(void)
 	}
 }
 
+void gic_spi_release_to_ns(size_t it)
+{
+	struct gic_data *gd = &gic_data;
+	size_t idx = it / NUM_INTS_PER_REG;
+	uint32_t mask = 1 << (it % NUM_INTS_PER_REG);
+
+	assert(it < gd->max_it && it >= GIC_SPI_BASE);
+	/* Assert it's already disabled */
+	assert(!gic_it_is_enabled(gd, it));
+	/* Assert it's secure to start with */
+	assert(!gic_it_get_group(gd, it));
+
+	gic_it_set_cpu_mask(gd, it, 0);
+	gic_it_set_prio(gd, it, GIC_SPI_PRI_NS_EL1);
+
+	/* Clear pending status */
+	io_write32(gd->gicd_base + GICD_ICPENDR(idx), mask);
+	/* Assign it to NS Group1 */
+	io_setbits32(gd->gicd_base + GICD_IGROUPR(idx), mask);
```

Comment:
I agree, since we're writing to a value register, not a set or clear bits register.
This function is in contrast to `void gic_init_donate_sgi_to_ns()` used post-boot when more than one core might be active in the secure world.

Reply:
done